src/renderer/components/Settings.tsx

```
@@ -0,0 +1,151 @@
+    } catch (error) {
+      console.error("Failed to select folder:", error);
+    } finally {
+      setIsScanning(false);
+    }
+  };
+  const handleScanFolder = async (folder: string) => {
+    setIsScanning(true);
+    console.log("scanning folder", folder);
+    const result = await window.electronAPI?.scanFolder(folder);
+    const newFolders = new Map(selectedFolders);
+    let date = result ? new Date() : null;
```

ESLint

[error] 'date' is never reassigned. Use 'const' instead.

(prefer-const)